### PR TITLE
qemu_img_measure_convert_image: add support for luks format

### PR DESCRIPTION
**Pull request by XueqiangWei**
Signed-off-by: Xueqiang Wei <xuwei@redhat.com>
ID: 1883250

### CODE REVIEW

**Comment by XueqiangWei** [comment]
Tested with luks, qcow2 and raw format, all passed.

(1/3) Host_RHEL.m8.u3.product_av.luks.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.qemu_img_measure.convert_image.convert_to_qcow2.q35: PASS (83.52 s)
 (2/3) Host_RHEL.m8.u3.product_av.luks.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.qemu_img_measure.convert_image.convert_to_raw.q35: PASS (71.22 s)
 (3/3) Host_RHEL.m8.u3.product_av.luks.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.qemu_img_measure.convert_image.convert_to_luks.q35: PASS (134.76 s)

 (1/3) Host_RHEL.m8.u3.product_av.raw.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.qemu_img_measure.convert_image.convert_to_qcow2.q35: PASS (31.57 s)
 (2/3) Host_RHEL.m8.u3.product_av.raw.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.qemu_img_measure.convert_image.convert_to_raw.q35: PASS (27.77 s)
 (3/3) Host_RHEL.m8.u3.product_av.raw.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.qemu_img_measure.convert_image.convert_to_luks.q35: PASS (90.32 s)

 (1/3) Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.qemu_img_measure.convert_image.convert_to_qcow2.q35: PASS (22.64 s)
 (2/3) Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.qemu_img_measure.convert_image.convert_to_raw.q35: PASS (28.43 s)
 (3/3) Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.qemu_img_measure.convert_image.convert_to_luks.q35: PASS (99.17 s)

**Comment by XueqiangWei** [comment]
@zhencliu Please help review it, many thanks.

**Comment by XueqiangWei** [comment]
depends on avocado-framework/avocado-vt/pull/2776

File: qemu/tests/qemu_img_measure_convert_image.py [hunk 1]
```diff
@@ -34,8 +35,14 @@ def _get_img_obj_and_params(tag):
 
     def _qemu_io(img, cmd):
         """Run qemu-io cmd to a given img."""
-        logging.info("Run qemu-io %s" % img.image_filename)
-        q = QemuIOSystem(test, params, img.image_filename)
+        image_filename = img.image_filename
+        logging.info("Run qemu-io %s" % image_filename)
+        if img.image_format == "luks":
+            image_secret_object = img._secret_objects[0]
```
Comment (zhencliu) [inline qemu/tests/qemu_img_measure_convert_image.py:41]: if qemu-io doesn't need the image's backings, we can use img._secret_objects[-1] here, or we have to define all secret objects

Comment (XueqiangWei) [reply qemu/tests/qemu_img_measure_convert_image.py:41]: img._secret_objects[-1]  is ok, I will update it. thanks

**Comment by XueqiangWei** [comment]
Test results:
(1/3) Host_RHEL.m8.u3.product_av.luks.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.qemu_img_measure.new_image.q35: PASS (57.70 s)
 (2/3) Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.qemu_img_measure.new_image.q35: PASS (38.34 s)
 (3/3) Host_RHEL.m8.u3.product_av.raw.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.qemu_img_measure.new_image.q35: PASS (38.58 s)

 (1/9) Host_RHEL.m8.u3.product_av.luks.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.qemu_img_measure.convert_image.convert_to_qcow2.q35: PASS (82.84 s)
 (2/9) Host_RHEL.m8.u3.product_av.luks.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.qemu_img_measure.convert_image.convert_to_raw.q35: PASS (69.14 s)
 (3/9) Host_RHEL.m8.u3.product_av.luks.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.qemu_img_measure.convert_image.convert_to_luks.q35: PASS (123.26 s)
 (4/9) Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.qemu_img_measure.convert_image.convert_to_qcow2.q35: PASS (26.20 s)
 (5/9) Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.qemu_img_measure.convert_image.convert_to_raw.q35: PASS (27.30 s)
 (6/9) Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.qemu_img_measure.convert_image.convert_to_luks.q35: PASS (93.14 s)
 (7/9) Host_RHEL.m8.u3.product_av.raw.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.qemu_img_measure.convert_image.convert_to_qcow2.q35: PASS (37.07 s)
 (8/9) Host_RHEL.m8.u3.product_av.raw.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.qemu_img_measure.convert_image.convert_to_raw.q35: PASS (25.67 s)
 (9/9) Host_RHEL.m8.u3.product_av.raw.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.qemu_img_measure.convert_image.convert_to_luks.q35: PASS (100.54 s)

**Comment by XueqiangWei** [comment]
@zhencliu Please help review it again. Many thanks.

File: qemu/tests/qemu_img_measure_convert_image.py [hunk 2]
```diff
@@ -34,8 +35,14 @@ def _get_img_obj_and_params(tag):
 
     def _qemu_io(img, cmd):
         """Run qemu-io cmd to a given img."""
-        logging.info("Run qemu-io %s" % img.image_filename)
-        q = QemuIOSystem(test, params, img.image_filename)
+        image_filename = img.image_filename
+        logging.info("Run qemu-io %s" % image_filename)
+        if img.image_format == "luks":
+            image_secret_object = img._secret_objects[-1]
```
Comment (zhencliu) [inline qemu/tests/qemu_img_measure_convert_image.py:41]: We might miss something here: _secret_objects is protected...
But, as this is a workaround for QemuIOSystem(luks image), and we don't want to make code changes for QemuIOSystem, so I think it's OK.

**Comment by zhencliu** [comment]
LGTM

**Comment by XueqiangWei** [comment]
Test results:
(1/3) Host_RHEL.m8.u3.product_av.luks.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.qemu_img_measure.new_image.q35: PASS (57.32 s)
(2/3) Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.qemu_img_measure.new_image.q35: PASS (38.28 s)
(3/3) Host_RHEL.m8.u3.product_av.raw.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.qemu_img_measure.new_image.q35: PASS (38.23 s)

(1/9) Host_RHEL.m8.u3.product_av.luks.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.qemu_img_measure.convert_image.convert_to_qcow2.q35: PASS (80.50 s)
(2/9) Host_RHEL.m8.u3.product_av.luks.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.qemu_img_measure.convert_image.convert_to_raw.q35: PASS (71.19 s)
(3/9) Host_RHEL.m8.u3.product_av.luks.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.qemu_img_measure.convert_image.convert_to_luks.q35: PASS (121.94 s)
(4/9) Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.qemu_img_measure.convert_image.convert_to_qcow2.q35: PASS (23.38 s)
(5/9) Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.qemu_img_measure.convert_image.convert_to_raw.q35: PASS (27.48 s)
(6/9) Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.qemu_img_measure.convert_image.convert_to_luks.q35: PASS (98.85 s)
(7/9) Host_RHEL.m8.u3.product_av.raw.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.qemu_img_measure.convert_image.convert_to_qcow2.q35: PASS (36.58 s)
(8/9) Host_RHEL.m8.u3.product_av.raw.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.qemu_img_measure.convert_image.convert_to_raw.q35: PASS (26.42 s)
(9/9) Host_RHEL.m8.u3.product_av.raw.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.qemu_img_measure.convert_image.convert_to_luks.q35: PASS (101.20 s)

**Comment by XueqiangWei** [comment]
@zhencliu vt's pullreq has been merged, please help review it. Many thanks.

**Pull request merged**